api/envoy/api/v2/cluster/circuit_breaker.proto

```diff
@@ -39,9 +55,22 @@ message CircuitBreakers {
     // upstream cluster. If not specified, the default is 1024.
     google.protobuf.UInt32Value max_requests = 4;
 
-    // The maximum number of parallel retries that Envoy will allow to the
-    // upstream cluster. If not specified, the default is 3.
-    google.protobuf.UInt32Value max_retries = 5;
+    // If neither max_retries or retry_budget is specified, a max_retries value
+    // of 3 is enforced.
+    oneof retry_threshold {
```

Comment:
We no longer allow an inline upgrade to oneof as it is a breaking change for some clients. I would drop the oneof for now and just verify in code that only one of them is set. @htuch @lizan do we have plans to add some type of automatic oneof upgrade annotation?

Reply:
It's not really necessary to have the oneof there since it's documented in several places that the budget will override the retry CB. I'll take care of that.

Reply:
@mattklein123 yes, but it's lower priority than rename and package movement, both of which we really need to make v3 sane. I think it wouldn't hurt to add the annotation now; basically something in https://github.com/cncf/udpa/blob/master/udpa/annotations/migrate.proto#L15 that says "add_to_oneof", with a string providing a  `oneof` as a target.